### PR TITLE
Scripts to check issue 1877

### PR DESCRIPTION
**Pull request by HSavynetska**
ATF Test Scripts to check [#1877](https://github.com/smartdevicelink/sdl_core/issues/1877)

This PR is **ready** for review.

### Summary
A test script to verify that issue is fixed

### ATF version
[develop](https://github.com/smartdevicelink/sdl_atf/tree/develop)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+local common = require('user_modules/sequences/actions')
+local runner = require('user_modules/script_runner')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Functions ]]
+local function activateApp(pAppId)
+    if not pAppId then pAppId = 1 end
+    local requestId = common.getHMIConnection():SendRequest("SDL.ActivateApp", { appID = 12345, fakeParam = "fakeParam" })
```
Comment (JackLivio) [inline test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua]: 12345 is not the app id used for the HMI representation.
```
application_manager/src/rpc_service_impl.cc:580 rpc_service::RPCServiceImpl::ConvertSOtoMessage: Convertion result: {
   "jsonrpc" : "2.0",
   "method" : "BasicCommunication.OnAppRegistered",
   "params" : {
      "application" : {
         "appID" : 1601923125,
         "appName" : "Test Application",
         "appType" : [ "NAVIGATION" ],
         "deviceInfo" : {
            "id" : "c17d9e6f098bf334afc8862a6cb543a48d85f885d38788b2a818662b239aa6c6",
            "isSDLAllowed" : true,
            "name" : "127.0.0.1:12345",
            "transportType" : "WIFI"
         },
         "hmiDisplayLanguageDesired" : "EN-US",
         "isMediaApplication" : true,
         "policyAppID" : "0000001",
         "requestSubType" : [],
         "requestType" : []
      },
      "priority" : "NONE"
   }
}
```

**Comment by dboltovskyi** [comment]
@JackLivio Please do not review this PR since it hasn't passed our internal review yet. Thank you.

File: test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+-- Description:
+-- SDL does not ignore invalid request/response from hmi and does not log internal error
+-- Precondition:
+-- 1) Core, HMI started.
+-- 2) App is registered on HMI.
+-- In case:
+-- 1) HMI sends ActivateApp with fake param and invalid appID param
+-- Expected result:
+-- 1) SDL cuts off fake parameters, log the corresponding error and ignore this response (request)
+-- Actual result:
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua]: Please remove actual result

File: test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua [hunk 3]
```diff
@@ -0,0 +1,44 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/smartdevicelink/sdl_core/issues/1877
+--
+-- Description:
+-- SDL does not ignore invalid request/response from hmi and does not log internal error
+-- Precondition:
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua]: Please remove precondition, it is not actual for script

File: test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+-- 1) Core, HMI started.
+-- 2) App is registered on HMI.
+-- In case:
+-- 1) HMI sends ActivateApp with fake param and invalid appID param
+-- Expected result:
+-- 1) SDL cuts off fake parameters, log the corresponding error and ignore this response (request)
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua]: Please update `Expected result` according to defect description.

File: test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+--[[ Required Shared libraries ]]
+local common = require('user_modules/sequences/actions')
+local runner = require('user_modules/script_runner')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Functions ]]
+local function activateApp(pAppId)
+    if not pAppId then pAppId = 1 end
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua]: Please remove pAppId from function, it is redundant.

File: test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua (hunk 1 above)
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua]: According to defect description the appId must bw with invalid value, please update.

File: test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    local requestId = common.getHMIConnection():SendRequest("SDL.ActivateApp", { appID = 12345, fakeParam = "fakeParam" })
+    common.getHMIConnection():ExpectResponse(requestId)
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1877_SDL_does_not_ignore_invalid_request_response_from_hmi_and_does_not_log_internal_error.lua]: Response with `INVALID_DATA` result code must be expected.

**Comment by dboltovskyi** [comment]
@JackLivio Script is ready for review

**Comment by dboltovskyi** [comment]
Since the original issue is closed due to duplication.
Case is implemented in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1962

**Pull request closed without merging**